R/zchunk_LB120.LC_GIS_R_LTgis_Yh_GLU.R

```diff
@@ -43,25 +43,33 @@ module_aglu_LB120.LC_GIS_R_LTgis_Yh_GLU <- function(command, ...) {
 
     # Perform computations
 
-    L100.Land_type_area_ha %>%
-      # Add data for GCAM region ID and GLU
-      left_join_error_no_match(distinct(iso_GCAM_regID, iso, .keep_all = TRUE), by = "iso") %>%
-      # Add vectors for land type ( SAGE, HYDE, and WDPA )
-      left_join_error_no_match(LDS_land_types, by = c("land_code" = "Category")) %>%
-      left_join(SAGE_LT, by = "LT_SAGE") %>%  # includes NAs
-      rename(LT_SAGE_5 = Land_Type) %>%
-      # Drop all rows with missing values (inland bodies of water)
-      na.omit %>%
-      # Reset WDPA classification to "Non-protected" where HYDE classification is cropland, pasture, or urban land
-      mutate(LT_WDPA = if_else(LT_HYDE!="Unmanaged", "Non-protected", "Unmanaged"),
-             Land_Type = LT_SAGE_5,
-             # These multi-tiered classifications will be used for C contents, but for all land cover processing, collapse into GCAM land types
-             Land_Type = if_else(LT_HYDE == "Cropland", "Cropland", Land_Type),
-             Land_Type = if_else(LT_HYDE == "Pasture", "Pasture", Land_Type),
-             Land_Type = if_else(LT_HYDE == "UrbanLand", "UrbanLand", Land_Type),
-             # Area in thousand square kilometers (bm2)
-             Area_bm2 =  value * CONV_HA_BM2) ->
-      L100.Land_type_area_ha
+    land.type <-
+        L100.Land_type_area_ha %>%
+          ## Add data for GCAM region ID and GLU
+          left_join_error_no_match(distinct(iso_GCAM_regID, iso, .keep_all = TRUE), by = "iso") %>%
+          ## Add vectors for land type ( SAGE, HYDE, and WDPA )
+          left_join_error_no_match(LDS_land_types, by = c("land_code" = "Category")) %>%
+          left_join(SAGE_LT, by = "LT_SAGE") %>%  # includes NAs
+          rename(LT_SAGE_5 = Land_Type) %>%
+          ## Drop all rows with missing values (inland bodies of water)
+          na.omit
+
+    ## Reset WDPA classification to "Non-protected" where HYDE classification
+    ## is cropland, pasture, or urban land
+    hyde <- land.type$LT_HYDE
+    ltype <- land.type$LT_SAGE_5
+
+    land.type$LT_WDPA <- replace(hyde, hyde != "Unmanaged", "Non-protected")
+
+    land.type$Land_Type <-
+        ltype %>%
+          replace(hyde=='Cropland', 'Cropland') %>%
```

Comment:
That's a little strange.  The problem seems to be with `if_else`, rather than mutate:
```
> x <- runif(1000000)
> df <- tibble(x=x, y=replace(x, x<0.1, NA))
> system.time({mutate(df, y = if_else(is.na(y), -1, y)) -> dfr1})
   user  system elapsed 
  0.196   0.016   0.212 
> system.time({mutate(df, y=replace(y, is.na(y), -1)) -> dfr3})
   user  system elapsed 
  0.005   0.000   0.005 
> system.time({tidyr::replace_na(df, list(y=-1)) -> dfr2})
   user  system elapsed 
  0.005   0.000   0.005 
```
Both `replace` + `mutate` and `replace_na` outperform `if_else` + `mutate` dramatically.  (Note this contradicts what I said above about `if_else` and `replace` having similar performance characteristics -- I must have been testing on vectors that were too small.)  It's hard to understand why `if_else` is behaving so badly.  Maybe I'll open an issue on their repository asking about it.  Note that in light of this, `if_else(condition(x), y, z)` is probably better written as
```
cond <- condition(x)
x <- replace(x, cond, y)
x <- replace(x, !cond, z)
```
Also, `replace_na` has the advantage that it can work on multiple columns with a different replacement value in each column, so that's a big win for clarity when it's applicable.